graphql_compiler/query_formatting/common.py

```diff
@@ -13,10 +20,89 @@
 # Public API
 ######
 
+def _check_is_string_value(value):
+    """Raise if the value is not a proper utf-8 string."""
+    if not isinstance(value, six.string_types):
+        if isinstance(value, bytes):  # should only happen in py3
+            value.decode('utf-8')  # decoding should not raise errors
+        else:
+            raise GraphQLInvalidArgumentError(u'Attempting to convert a non-string into a string: '
+                                              u'{}'.format(value))
+
+
+def validate_argument_type(expected_type, value):
+    """Check if the value is appropriate for the type and usable in any of our backends.
+
+    Backends are the database languages we have the ability to compile to, like OrientDB Match,
+    Gramlin, or SQLAlchemy. This function should be stricter than the validation done by any
+    specific backend. That way code that passes validation can be compiled to any backend.
+
+    Args:
+        expected_type: GraphQL type we expect
```

Comment:
Is the a case class for the GraphQL types we could use here to be more specific?

Reply:
It's complicated https://github.com/graphql-python/graphql-core/blob/master/graphql/type/definition.py#L21

Reply:
Consider at least using GraphQLType.